test/prov_recurse_config_test.c

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2021 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#include <openssl/evp.h>
+#include <openssl/err.h>
+#include <openssl/conferr.h>
+#include "testutil.h"
+
+static char *configfile = NULL;
+
+/*
+ * Load a provider config with a recursive section and check
+ * for errors after the fact to ensure we caught and aborted the
+ * load
+ */
+
+static int test_recursive_config(void)
```

Comment:
Wouldn't this make more sense included in the existing test file for testing provider configuration (i.e. prov_config_test.c). It seems excessive to have a whole new test file just for this.

Reply:
my thought was that the intention was to have a recursive config result in an error during config load.  As such adding a recursive config element to an existing config would introduce issues with whatever other tests were using that config.  Is there a better way to do this?

Reply:
If you load the config file into a new libctx then it will only impact that libctx.

Reply:
...which actually you are doing anyway in this test!

Reply:
i've moded the test to test_provider_config